src/Convert/numpy.jl

```diff
@@ -54,4 +114,14 @@ function init_numpy()
         iscomplex && pyconvert_add_rule(name, Complex, rule)
         isnumber && pyconvert_add_rule(name, Number, rule)
     end
+
+    priority = PYCONVERT_PRIORITY_ARRAY
+    pyconvert_add_rule("numpy:datetime64", DateTime, pyconvert_rule_datetime64, priority)
+    for T in (CompoundPeriod, Year, Month, Day, Hour, Minute, Second, Millisecond, Microsecond, Nanosecond, Week)
+        pyconvert_add_rule("numpy:timedelta64", T, pyconvert_rule_timedelta64, priority)
+    end
```

Comment:
Since Julia is unlikely to unroll this loop, you should use `Base.Cartesian.@nexpr` here to avoid dynamic dispatch.

Reply:
Tried my best, but I'm not sure how to test whether this will speed up things